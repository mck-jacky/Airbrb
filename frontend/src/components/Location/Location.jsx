import React from 'react';
import PropTypes from 'prop-types';

const Location = ({ address }) => {
  return (
    <div className="mapouter" data-testid='location'>
      <div className="gmap_canvas">
        <iframe
        width="100%"
        height="500"
        id="gmap_canvas"
        src={'https://maps.google.com/maps?q=' + address + '&t=&z=11&ie=UTF8&iwloc=&output=embed'}
        frameBorder="0"
        scrolling="no"
        marginHeight="0"
        marginWidth="0">
        </iframe>
      </div>
    </div>
  );
};

Location.propTypes = {
  address: PropTypes.string
}

export default Location;
