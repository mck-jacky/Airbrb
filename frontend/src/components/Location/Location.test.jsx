import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow'
import Location from './Location'

afterEach(() => {
  cleanup();
})

describe('Location', () => {
  it('is in the document', () => {
    render(<Location address='UNSW,Sydney,NSW' />)
    const component = screen.getByTestId('location')
    expect(component).toBeInTheDocument();
  })

  it('has iframe', () => {
    render(<Location address='UNSW,Sydney,NSW' />)
    const component = screen.getByTestId('location')
    expect(component).toContainHTML('<iframe ')
  })

  it('has map src', () => {
    render(<Location address='UNSW,Sydney,NSW' />)
    const component = screen.getByTestId('location')
    expect(component).toContainHTML('src="https://maps.google.com/maps?q=UNSW,Sydney,NSW')
  })

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<Location address='UNSW,Sydney,NSW' />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
