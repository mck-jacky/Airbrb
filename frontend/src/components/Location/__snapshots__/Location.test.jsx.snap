// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location has to match the snapshot 1`] = `
<div
  className="mapouter"
  data-testid="location"
>
  <div
    className="gmap_canvas"
  >
    <iframe
      frameBorder="0"
      height="500"
      id="gmap_canvas"
      marginHeight="0"
      marginWidth="0"
      scrolling="no"
      src="https://maps.google.com/maps?q=UNSW,Sydney,NSW&t=&z=11&ie=UTF8&iwloc=&output=embed"
      width="100%"
    />
  </div>
</div>
`;
